feat: add social auth docs (#1014)
alchemyplatform/aa-sdk: site/pages/react/authenticate-users.mdx

}
```

> ### Social Login

Reviewer: Can you please add an instruction that tells the user they have to go to the Dashboard to enable social auth?

and please explain what effect the fields in the dashboard have on the end product.

> You can allow uers to sign in with a Oauth provider like Google, Facebook, or Auth0. There are two login flows that are possible: the redirect flow goes to the auth provider's login page and the popup flow has the auth provider login screen to open as a popup.

Reviewer: ```suggestion
Allow users to sign in with an OAuth provider like Google, Facebook, or Auth0. This will trigger a popup window or a redirect to the auth provider's login page.
```

- write declaratively. "Allow users" rather than "you can allow"
- 'an' before OAuth
- OAuth has a capital A

```tsx twoslash
import React from "react";
import {
  type UseAuthenticateResult,
  useAuthenticate,
} from "@account-kit/react";

// This examples uses social login, specifically Google Sign-In
export default function MyLoginPage() {
  const { authenticate, isPending } = useAuthenticate();

  return (
    <div>
      <button onClick={() => authenticate({ // redirect login flow
        type: "oauth",
        authProviderId: "google",
        mode: "redirect",
        redirectUrl: "/", // After logging in, redirect to the index page
      })}>
        Login (Redirect)
      </button>
      <button onClick={() => authenticate({ // popup login flow
        type: "oauth",
        authProviderId: "google",
        mode: "popup",
      })}>
        Login (Popup)

>       </button>

Reviewer 2: If we're demonstrating popup-based login, we need to make clear that users need to set `enablePopupLogin` in their config if they want to do this. This is a required step that we should mention anywhere we introduce popup login.
